photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,282 @@
+    def do_photometry(self):
+        pass
+
+
+class NStarPSFPhotometry(PSFPhotometryBase):
+    """
+    This is an implementation of the NSTAR algorithm proposed by Stetson
+    (1987) to perform point spread function photometry in crowded fields.
+
+    This class basically implements the loop FIND, GROUP, NSTAR,
+    SUBTRACT, FIND until no more stars are detected.
+    """
+
+    def __init__(self, find, group, bkg, psf, fitter, niters, fitshape):
+        """
+        Attributes
+        ----------
+        find : an instance of any StarFinderBase subclasses
+        group : an instance of any GroupStarsBase subclasses
+        bkg : an instance of any BackgroundBase2D (?) subclasses
+        psf : Fittable2DModel instance
+        fitter : Fitter instance
+        niters : int
+            number of iterations for the loop FIND, GROUP, SUBTRACT, NSTAR
+        fitshape : array-like
+            rectangular shape around the center of a star which will be used
+            to collect the data to do the fitting
+        """
+
+        self.find = find
+        self.group = group
+        self.bkg = bkg
+        self.psf = psf
+        self.fitter = fitter
+        self.niters = niters
+        self.fitshape = fitshape
+
+        @property
+        def niters(self):
+            return self._niters
+        @niters.setter
+        def niters(self, niters):
+            if isinstance(niters, int) and niters > 0:
+                self._niters = niters
+            else:
+                raise ValueError('niters is not defined properly, '
+                                 'received niters = {}'.format(niters))
+        @property
+        def fitshape(self):
+            return self._fitshape
+        @fitshape.setter
+        def fitshape(self, fitshape):
+            fitshape = np.asarray(fitshape)
+            if len(fitshape) == 2 and np.all(fitshape) > 0:
+                self._fitshape = fitshape
+            else:
+                raise ValueError('fitshape is not defined properly, '
+                                 'received fitshape = {}'.format(fitshape))
+
+    def __call__(self, image):
+        """
+        Parameters
+        ----------
+        image : array-like, ImageHDU, HDUList
+            image to perform photometry
+        
+        Returns
+        -------
+        outtab : astropy.table.Table
+            Table with the photometry results, i.e., centroids and flux
+            estimations.
+        residual_image : array-like, ImageHDU, HDUList
+            Residual image calculated by subtracting the fitted sources
+            and the original image.
+        """
+
+        return self.do_photometry(image)
+
+    def _nstar(self, image, star_groups):
+        """
+        Fit, as appropriate, a compound or single model to a given `groups` of
+        stars. Groups are fitted sequentially from the smallest to the biggest. In
+        each iteration, `image` is subtracted by the previous fitted group. 
```

Comment:
Use double ticks for all the parameter names (``image`` here), otherwise sphinx will fail (when trying to resolve them as links)